src/components/Modal/Modal.js

```diff
@@ -0,0 +1,183 @@
+import React, { Component } from 'react';
+import { TextField } from '@mui/material';
+import getSymbolFromCurrency from 'currency-symbol-map';
+
+import {
+  CloseButton,
+  ModalButton,
+  ModalOverlay,
+  ModalContainer,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalTitle,
+  RightContent,
+  LeftContent,
+  CoinImageContainer,
+  CoinImage,
+  CoinNameText,
+  StyledDropdown,
+  ModalButtons,
+  StyledPaper,
+  StyledListbox,
+  StyledDate,
+  StyledCurrency,
+  StyledButton
+} from './Modal.styles';
+
+export class Modal extends Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef(null);
+  }
```

Comment:
ref = React.createRef(null); same as you do with the state, no need for a contructor